torch/_dynamo/variables/lazy.py

```diff
@@ -80,6 +80,14 @@ def clone(self, **kwargs: Any) -> VariableTracker:
             self.realize()
         return VariableTracker.clone(self.unwrap(), **kwargs)
 
+    def peek_type(self) -> type[Any]:
+        assert not self.is_realized()
+        return type(self._cache.value)
```

Comment:
Good idea! I was thinking if we can somehow use this for `isinstance`? Today `isinstance` on lazy variable realizes them.